nifi-nar-bundles/nifi-pgp-bundle/nifi-pgp-processors/src/main/java/org/apache/nifi/processors/pgp/EncryptContentPGP.java

```diff
@@ -0,0 +1,379 @@
+                final ValidationResult result = new ValidationResult.Builder()
+                        .valid(false)
+                        .subject(getClass().getSimpleName())
+                        .explanation(explanation)
+                        .build();
+                results.add(result);
+            }
+        }
+
+        return results;
+    }
+
+    private StreamCallback getEncryptStreamCallback(final ProcessContext context, final FlowFile flowFile,
+                                                    final SymmetricKeyAlgorithm symmetricKeyAlgorithm,
+                                                    final CompressionAlgorithm compressionAlgorithm,
+                                                    final FileEncoding fileEncoding) {
+        final SecureRandom secureRandom = new SecureRandom();
+        final PGPDataEncryptorBuilder dataEncryptorBuilder = new BcPGPDataEncryptorBuilder(symmetricKeyAlgorithm.getId())
+                .setSecureRandom(secureRandom)
+                .setWithIntegrityPacket(ENCRYPTION_INTEGRITY_PACKET_ENABLED);
+        final PGPEncryptedDataGenerator encryptedDataGenerator = new PGPEncryptedDataGenerator(dataEncryptorBuilder);
+        final List<PGPKeyEncryptionMethodGenerator> methodGenerators = getEncryptionMethodGenerators(context, flowFile, secureRandom);
+        methodGenerators.forEach(encryptedDataGenerator::addMethod);
+
+        final String filename = flowFile.getAttribute(CoreAttributes.FILENAME.key());
+        return new EncryptStreamCallback(filename, fileEncoding, encryptedDataGenerator, compressionAlgorithm);
+    }
+
+    private List<PGPKeyEncryptionMethodGenerator> getEncryptionMethodGenerators(final ProcessContext context,
+                                                                                final FlowFile flowFile,
+                                                                                final SecureRandom secureRandom) {
+        final List<PGPKeyEncryptionMethodGenerator> generators = new ArrayList<>();
+
+        final PropertyValue passphraseProperty = context.getProperty(PASSPHRASE);
+        if (passphraseProperty.isSet()) {
+            final char[] passphrase = passphraseProperty.getValue().toCharArray();
+            generators.add(new JcePBEKeyEncryptionMethodGenerator(passphrase).setSecureRandom(secureRandom));
+        }
+
+        final PropertyValue publicKeySearchProperty = context.getProperty(PUBLIC_KEY_SEARCH);
+        if (publicKeySearchProperty.isSet()) {
+            final String publicKeySearch = publicKeySearchProperty.evaluateAttributeExpressions(flowFile).getValue();
+            getLogger().debug("Public Key Search [{}]", publicKeySearch);
+
+            final PGPPublicKeyService publicKeyService = context.getProperty(PUBLIC_KEY_SERVICE).asControllerService(PGPPublicKeyService.class);
+            final Optional<PGPPublicKey> optionalPublicKey = publicKeyService.findPublicKey(publicKeySearch);
```

Comment:
This can throw an NPE if you do the following:

1. Configure the processor with a Public Key Service
2. Specify a Public Key Search value
3. Set the Public Key Service to None
4. Specify a Passphrase
5. Run a flow file through the processor (this gives you the NPE)

Reply:
Thanks for pointing out the problem and providing the steps to reproduce!  Will introduce additional checking to ensure the service is configured before calling `findPublicKey()`